File: demos/tx/theta-tx-demo/src/main/java/org/theta/types/TxInput.java
package org.theta.types;

import java.math.BigInteger;

public final class TxInput {

    public byte[] address;
    public Coins  coins;
    public long   sequence;
    public String signature;

    public TxInput(byte[] address, BigInteger thetaWei, BigInteger gammaWei, long sequence) {
        this.address = address;
        this.coins = new Coins(thetaWei, gammaWei);
        this.sequence = sequence;
    }

    public void setSignature(String signature) {
        this.signature = signature;
    }
}

File: demos/tx/javascript-demo/theta-tx-demo/src/tx/types/txinput.js
import {Coins} from './coins'
import Bytes from 'eth-lib/lib/bytes';

export class TxInput{
    constructor(address, thetaWei, tfuelWei, sequence) {
        this.address = address;
        this.coins = new Coins(thetaWei, tfuelWei);
        this.sequence = sequence;
        this.signature = "";
    }

    setSignature(signature) {
        this.signature = signature;
    }

    rlpInput(){
        let rplInput = [
            this.address.toLowerCase(),
            this.coins.rlpInput(),
            Bytes.fromNumber(this.sequence),
            this.signature
        ];

        return rplInput;
    }
}